### PR TITLE
[DOC, MAINT] Re-organization of pages in the docu web

### PR DESCRIPTION
**Pull request by juangpc**
working here: https://juangpc.github.io/mne-cpp/

I intend to make it easier to: 

- segregate between users and developers. This is. a user doesn't need to know what a "binary file" is, an executable, etc... nor a development version, etc... It is just a program or an application that they use. I also delete 1 click from the home page to the download page. It doesn't need to know what is "linking" nor "static" or "dynamic" linking. Or what is "webassembly". I think they just need to know something more abstract like "MNE on the web". Therefore I implicitly create a distinction between *USER*, which belongs to "Download" link and navigation, and perhaps the "Documentation" link also. And *DEVELOPER* which should understand more specific concepts only related to the underpinning technology shared here.
- We use "documentation", here in the development. In the user manual, in our conversations, in the grant documents. So I would propose to call it that, instead of "learn".
- We have now. "Build from source" in the download page. The only reason why that page could be under Download is when a user of an unsupported architecture wants to USE our sw. In that case, it would be a USER that needs to Build from source. But, on every other case, anybody wanting to build the project is probably a developer. For that reason I propose to consider "Build from source" documentation an information page which should be understood only within the DEVELOPER context. 
- I've changed the order of a few menus to try to make it more coherent. 
Let's see what you think.

### CODE REVIEW

**Comment by juangpc** [comment]
Did the changes we commented. 
You guys can have a look now.

**Comment by LorenzE** [comment]
Looking good! From myl experience rearranging .md files to other/new folders easily breaks internal links. For example the link to the "Build from source" in the "Download > Development Versions" page is currently not working because the build guide was moved to the "Development" section.

**Comment by juangpc** [comment]
I've created a new file in the sample dataset folder. It is an file named "download_sample_data.html". It is cross-platform and it contains an html page with a single line javascript instruction to go to our documentation webpage: [Sample Dataset](https://juangpc.github.io/mne-cpp/pages/download/sample_data.html)).

I've also modified the markdown file README.md so that it shows: 

image

Both changes are intended to "delegate" whatever instructions we give to only one point: our doucmentation webpage. If everything is pointing there, whenever, if ever, we change something, as long as we keep the url, users will not be confused. And we only need to maintain the documentation webpage.

## IMPORTANT
Right before this PR gets accepted, we need to change the urls to point to the actual mne-cpp webpage, not the one they are pointing at right now, for demo pursposes (juangpc).

**Comment by juangpc** [comment]
> Looking good! From myl experience rearranging .md files to other/new folders easily breaks internal links. For example the link to the "Build from source" in the "Download > Development Versions" page is currently not working because the build guide was moved to the "Development" section.

fixed in last commit

**Comment by juangpc** [comment]
Checking all the links for 404s...

We have a total of 303 links in the website. 
Failing now.... 15 ☹️ 
1. in _config.yml there is a reference to images/index.svg and there is no such file.... That's (i believe) an image that should be at the top left corner of the web. Testing with the logo of MNE-CPP.

**Comment by juangpc** [comment]
Hi again, 
I'm showing that almost half the broken link have were actually broken before this PR, so that means that we have about 6~ broken links right now. So if we drop this pr, we must remember to update the actual master we have right now.

**Comment by juangpc** [comment]
There is a broken link in the publications list: 

Dinh, C.; Luessi, M.; Sun, L.; Haueisen, J.; Hämäläinen, M.S.: MNE-X: MEG/EEG Real-Time Acquisition, Real-Time Processing, and Real-Time Source Localization Framework. Biomedizinische Technik (Berl.) (2013), 58, DOI: 10.1515/bmt-2013-4184

This publication's link is a 404. Could someone help me find a suitable link? @LorenzE @chdinh ?

**Comment by RDoerfel** [comment]
> There is a broken link in the publications list of the MNE-CPP website:
> 
> see: https://mne-cpp.github.io/publication.html
> 
> Dinh, C.; Luessi, M.; Sun, L.; Haueisen, J.; Hämäläinen, M.S.: MNE-X: MEG/EEG Real-Time Acquisition, Real-Time Processing, and Real-Time Source Localization Framework. Biomedizinische Technik (Berl.) (2013), 58, DOI: 10.1515/bmt-2013-4184
> 
> This publication's link is a 404. Could someone help me find a suitable link? @LorenzE @chdinh ?


[doi.org/10.1515/bmt-2013-4184](doi.org/10.1515/bmt-2013-4184) should solve it I think. 

And it seems like many of the links are leading to the same publication (number 3/4/5/6). 

Further, if you want to open the links in a new tab you would have to add: 
`{:target="_blank" rel="noopener"}`
to the links. See #456.

**Comment by juangpc** [comment]
> > There is a broken link in the publications list of the MNE-CPP website:
> > see: https://mne-cpp.github.io/publication.html
> > Dinh, C.; Luessi, M.; Sun, L.; Haueisen, J.; Hämäläinen, M.S.: MNE-X: MEG/EEG Real-Time Acquisition, Real-Time Processing, and Real-Time Source Localization Framework. Biomedizinische Technik (Berl.) (2013), 58, DOI: 10.1515/bmt-2013-4184
> > This publication's link is a 404. Could someone help me find a suitable link? @LorenzE @chdinh ?
> 
> [doi.org/10.1515/bmt-2013-4184](doi.org/10.1515/bmt-2013-4184) should solve it I think.
> 
> And it seems like many of the links are leading to the same publication (number 3/4/5/6).
> 
> Further, if you want to open the links in a new tab you would have to add:
> `{:target="_blank" rel="noopener"}`
> to the links. See #456.
> 
> And I like the new arrangement, it is getting more and more intuitive ;)

Added!

File: doc/gh-pages/overview.md [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-nav_order: 2
+nav_order: 1
 ---
 
+# The MNE-CPP Project
+
 ## What is MNE?
 
-MNE-CPP is part of MNE, a tool suite for neurophysiology encompassing various projects that draw from MNE-C, originally written by Matti Hämäläinen (documented [here](https://mne.tools/mne-c-manual/MNE-manual-2.7.3.pdf))
+MNE-CPP is part of [MNE Project](https://mne.tools/stable/index.html){:target="_blank" rel="noopener"}, a set of tools for neurophysiology analysis encompassing various projects that sprawn from MNE-C, originally written by Matti Hämäläinen.
```
Comment (gabrielbmotta) [inline doc/gh-pages/overview.md:11]: I'm not sure if 'sprawn' is a word. Maybe we could use 'originate', 'spawn, 'take inspiration',

File: doc/gh-pages/pages/development/development.md [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
 has_children: true
-nav_order: 5
+nav_order: 4
 ---
-# Develop
+# How to contribute to the MNE-CPP Project
+
+## Welcome aboard!
+
+Before you start, there is a few basic concepts that should be understood. We use a variety of different tools to develop and maintain MNE-CPP.
```
Comment (gabrielbmotta) [inline doc/gh-pages/pages/development/development.md:10]: Should be 'there are a few basic concepts'

File: doc/gh-pages/pages/development/development.md [hunk 3]
```diff
@@ -1,10 +1,39 @@
...
+
+## Before you start
+
+Learn C++ and the [Qt](https://www.qt.io/){:target="_blank" rel="noopener"}.
```
Comment (gabrielbmotta) [inline doc/gh-pages/pages/development/development.md:14]: Possibly could change from 'the Qt' to just 'Qt' or 'the Qt Framework'

File: doc/gh-pages/pages/development/development.md [hunk 4]
```diff
@@ -1,10 +1,39 @@
...
+
+## Welcome aboard!
+
+Before you start, there is a few basic concepts that should be understood. We use a variety of different tools to develop and maintain MNE-CPP.
+
+## Before you start
+
+Learn C++ and the [Qt](https://www.qt.io/){:target="_blank" rel="noopener"}. 
+
+We mainly use C++ programming language and Qt, for development purposes. We are very happy with the choice, but these technologies can be quite tricky. There are plenty of fantastic resources out there to be able to learn both, but if you feel stuck, go ahead and ask us. Since you intend to become a contributor, we recommend you to start using Github and open an issue in the MNE-CPP project, in order to communicate with the development team.
```
Comment (gabrielbmotta) [inline doc/gh-pages/pages/development/development.md:16]: Do we want the point of contact with us to be through opening an issue? maybe this is a good time to implement the 'github discussions' feature @LorenzE has been mentioning.

File: doc/gh-pages/pages/development/development.md [hunk 5]
```diff
@@ -1,10 +1,39 @@
...
+
+Below are some resources to get you started developing within MNE-CPP. Depending on which application you intend to develop for, you will find specific resources in this section. You can use use our [API documentation](https://mne-cpp.github.io/doxygen-api/annotated.html){:target="_blank" rel="noopener"} to guide you while understanding the responsabilities of each class.
+
+## Contribute
+
+It is a good idea to go through the contribution tutorial and make sure that you can build from source. Understand the continious integration pipeline we have in place and how to write a unit test. Make sure that your changes don't break the project compilation before submitting your contribution to the project itself.
```
Comment (gabrielbmotta) [inline doc/gh-pages/pages/development/development.md:34]: Maybe we can point them to the test script so they can test if their changes break the ci

File: doc/gh-pages/pages/download/development_versions.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+---
+title: Development Versions
+parent: Download
+nav_order: 1
+---
+# Development Versions
+
+For older releases please visit our [Github release page](https://github.com/mne-tools/mne-cpp/releases){:target="_blank" rel="noopener"}. If you want to checkout more recently added features, you can download the development versions of the application.
```
Comment (gabrielbmotta) [inline doc/gh-pages/pages/download/development_versions.md:8]: 'Checkout' should be 'check out', unless this is a git pun, in which case i'm all for it.

File: doc/gh-pages/pages/download/download.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+### 3. Select the tool you wish to use from the `bin` folder.
+
+In the `bin` folder,  you will find:
+  - A real-time analysis application: [MNE-Scan](../documentation/scan.md).
+  - An offline visualization/analysis application: [MNE Analyze](../documentation/analyze.md).
+  - An anonymization/deidentification tool for FIF files: [MNE Anonymize](../documentation/anonymize.md).
+
+---
+
+Using the latest stable release is recommended for users who wish to use MNE-CPP's command line tools and GUIs. But if you want to checkout the latest features, try downloading the [development versions](../development/development.md).
```
Comment (gabrielbmotta) [inline doc/gh-pages/pages/download/download.md:41]: 'Checkout' should be 'check out', unless this is a git pun, in which case i'm all for it.

File: doc/gh-pages/overview.md (hunk 1 above)
Comment (gabrielbmotta) [reply doc/gh-pages/overview.md:11]: sentence might work better with 'the MNE Project' instead of just 'MNE project'

**Review by gabrielbmotta — CHANGES_REQUESTED** [review]

File: doc/gh-pages/pages/download/download.md (hunk 7 above)
Comment (juangpc) [reply doc/gh-pages/pages/download/download.md:41]: Thank you very much.

File: doc/gh-pages/pages/download/development_versions.md (hunk 6 above)
Comment (juangpc) [reply doc/gh-pages/pages/download/development_versions.md:8]: It was not. Perhaps it should be. But just bad spelling.

File: doc/gh-pages/pages/development/development.md (hunk 5 above)
Comment (juangpc) [reply doc/gh-pages/pages/development/development.md]: done.

File: doc/gh-pages/pages/development/development.md (hunk 4 above)
Comment (juangpc) [reply doc/gh-pages/pages/development/development.md:16]: Agreed. I agree a lot. 
Let's do that.

File: doc/gh-pages/pages/development/development.md (hunk 3 above)
Comment (juangpc) [reply doc/gh-pages/pages/development/development.md]: Liking "the Qt Framework".

File: doc/gh-pages/pages/development/development.md (hunk 2 above)
Comment (juangpc) [reply doc/gh-pages/pages/development/development.md]: Thank you.

File: doc/gh-pages/overview.md (hunk 1 above)
Comment (juangpc) [reply doc/gh-pages/overview.md]: i don't think is a word. 😄 
Done

**Comment by juangpc** [comment]
last push addresses all the mistakes underlined by @gabrielbmotta. Thank you. This is looking better.

File: doc/gh-pages/overview.md [hunk 8]
```diff
@@ -8,7 +8,7 @@ nav_order: 1
 
 ## What is MNE?
 
-MNE-CPP is part of [MNE Project](https://mne.tools/stable/index.html){:target="_blank" rel="noopener"}, a set of tools for neurophysiology analysis encompassing various projects that sprawn from MNE-C, originally written by Matti Hämäläinen.
+The MNE-CPP is part of [MNE Project](https://mne.tools/stable/index.html){:target="_blank" rel="noopener"}, a set of tools for neurophysiology analysis encompassing various projects that spawn from MNE-C, originally written by Matti Hämäläinen.
```
Comment (gabrielbmotta) [inline doc/gh-pages/overview.md:11]: MNE-CPP was fine as is, the issue with 'the' was with 'MNE Project'. 'MNE-CPP is part of the MNE Project' or MNE-CPP is part of the MNE community' or maybe 'MNE-CPP is part of MNE'

Comment (juangpc) [reply doc/gh-pages/overview.md:11]: Doh!

Comment (juangpc) [reply doc/gh-pages/overview.md:11]: Done.

**Comment by LorenzE** [comment]
I really like it! Some findings:

- Download > MNE On the Browser -> There is information about the anonymizer here. Is this wanted? Also, the font is too big.
- The Development > Contribute page is empty. I think the content was moved to Development page. This is fine but maybe we should add some basic information under Development > Contribute again?
- Same as above for Development > Continuous Integration
- Download "But if you want to check out the latest features, try downloading the development versions." The included does not link to the download section for development versions.

**Comment by LorenzE** [comment]
Just wondering: was the last commit including code changes pushed to the wrong branch?

**Comment by juangpc** [comment]
> Download > MNE On the Browser -> There is information about the anonymizer here. Is this wanted? Also, the font is too big.

I believe we do want it there because already we've seen how users could be confused whether the information is travelling to a server or not. Although, not only for anonymizer, but for all wasm applications. So yes. 

> The Development > Contribute page is empty. I think the content was moved to Development page. This is fine but maybe we should add some basic information under Development > Contribute again?

I agree. Basic info added.

> Same as above for Development > Continuous Integration

I agree. Added.

> Download "But if you want to check out the latest features, try downloading the development versions." The included does not link to the download section for development versions.

Done.

>Just wondering: was the last commit including code changes pushed to the wrong branch?

While going through the plugin documentation I noticed the code and changed it. I agree this could be a separate pr. I reverted it.

File: applications/mne_analyze/plugins/annotationmanager/annotationmanager.cpp [hunk 9]
```diff
@@ -76,8 +76,7 @@ AnnotationManager::~AnnotationManager()
 
 QSharedPointer<AbstractPlugin> AnnotationManager::clone() const
 {
-    QSharedPointer<AnnotationManager> pAnnotationManagerClone(new AnnotationManager);
-    return pAnnotationManagerClone;
+    return QSharedPointer<AnnotationManager> (new AnnotationManager);
```
Comment (gabrielbmotta) [inline applications/mne_analyze/plugins/annotationmanager/annotationmanager.cpp:79]: These seem to be changes that fit the Event pr, were these mistakenly pushed here?

**Review by gabrielbmotta — CHANGES_REQUESTED** [review]

**Comment by gabrielbmotta** [comment]
A lot of the pages under Development/MNE Scan fall more on the documentation side of things, would those be better in that section? Also the link to the mne scan plugin creation guide pdf doesn't seem to be working (in either version of the website).

**Review by gabrielbmotta — APPROVED** [review]

**Pull request merged**